hikari/api/rest.py

```diff
@@ -8282,3 +8283,332 @@ def fetch_scheduled_event_users(
+        hikari.errors.InternalServerError
+            If an internal error occurs on Discord while handling the request.
+        """
+
+    @abc.abstractmethod
+    async def fetch_auto_mod_rule(
+        self,
+        guild: snowflakes.SnowflakeishOr[guilds.PartialGuild],
+        rule: snowflakes.SnowflakeishOr[auto_mod.AutoModRule],
+        /,
+    ) -> auto_mod.AutoModRule:
+        """Fetch a auto-moderation rule.
+
+        Parameters
+        ----------
+        guild : hikari.snowflakes.SnowflakeishOr[hikari.guilds.PartialGuild]
+            Object or ID of the guild to fetch the auto moderation rules of.
+        rule : hikari.snowflakes.SnowflakeishOr[hikari.auto_mod.AutoModRule]
+            Object or ID of the auto moderation rule to fetch.
+
+        Returns
+        -------
+        hiakri.auto_mod.AutoModRule
+            The fetched auto moderation rule.
+
+        Raises
+        ------
+        hikari.errors.BadRequestError
+            If any of the fields that are passed have an invalid value.
+        hikari.errors.ForbiddenError
+            If you are missing the `MANAGE_GUILD` permission.
+        hikari.errors.UnauthorizedError
+            If you are unauthorized to make the request (invalid/missing token).
+        hikari.errors.NotFoundError
+            If the guild or rule was not found.
+        hikari.errors.RateLimitTooLongError
+            Raised in the event that a rate limit occurs that is
+            longer than `max_rate_limit` when making a request.
+        hikari.errors.RateLimitedError
+            Usually, Hikari will handle and retry on hitting
+            rate-limits automatically. This includes most bucket-specific
+            rate-limits and global rate-limits. In some rare edge cases,
+            however, Discord implements other undocumented rules for
+            rate-limiting, such as limits per attribute. These cannot be
+            detected or handled normally by Hikari due to their undocumented
+            nature, and will trigger this exception if they occur.
+        hikari.errors.InternalServerError
+            If an internal error occurs on Discord while handling the request.
+        """
+
+    @abc.abstractmethod
+    async def create_auto_mod_rule(
+        self,
+        guild: snowflakes.SnowflakeishOr[guilds.PartialGuild],
+        /,
+        name: str,
+        event_type: typing.Union[int, auto_mod.AutoModEventType],
+        trigger_type: typing.Union[int, auto_mod.AutoModTriggerType],
+        actions: typing.Sequence[auto_mod.PartialAutoModAction],
+        allow_list: undefined.UndefinedOr[typing.Sequence[str]] = undefined.UNDEFINED,
+        keyword_filter: undefined.UndefinedOr[typing.Sequence[str]] = undefined.UNDEFINED,
+        presets: undefined.UndefinedOr[
+            typing.Sequence[typing.Union[int, auto_mod.AutoModKeywordPresetType]]
+        ] = undefined.UNDEFINED,
+        enabled: undefined.UndefinedOr[bool] = True,
```

Comment:
Would be better UX to take subclasses of `PartialAutoModTrigger` here instead of their config fields but that'd add complexity impl wise and idk if you want to keep these functions as flat as possible or not

Reply:
Since this didn't really get any comment i'ma just switch to that approach